packages/light-apps/src/AddAccount/AddAccount.tsx

```diff
@@ -15,49 +14,57 @@ import { Container } from 'semantic-ui-react';
 interface Props extends RouteComponentProps { }
 
 type State = {
-  blockNumber?: BlockNumber,
-  renameModalOpen: boolean,
-  backupModalOpen: boolean,
-  forgetModalOpen: boolean,
-  error?: string,
-  success?: string,
-  password: string,
-  name: string
-  newName: string
+  activeTab: string
 };
 
 export class AddAccount extends React.PureComponent<Props, State> {
   static contextType = AppContext;
 
+  state: State = {
+    activeTab: 'generate'
+  };
+
   context!: React.ContextType<typeof AppContext>; // http://bit.ly/typescript-and-react-context
 
+  componentWillReceiveProps (nextProps: any) {
```

Comment:
I'd advise against using this, because 1/ it's deprecated, and 2/ setState inside componentWillReceiveProps adds an additional unnecessary component render.

Here, i would just create a function `getActiveTab = () => this.props.location.pathname.split('/')[3]`, and call it from inside render(). I don't think there's need for state.

(I know that we use componentWillReceiveProps in `<Subscribe>`, but that's copy-pasted code, and hopefully will be replaced by hooks like you did in InputFile).